includes/admin/settings/register-settings.php

```diff
@@ -29,6 +29,35 @@ function edd_get_option( $key = '', $default = false ) {
 }
 
 /**
+ * Update an option
+ *
+ * Updates an edd setting value in both the db and the global variable.
+ *
+ * @since 2.3
+ * @return boolean True if updated, false if not.
+ */
+function edd_update_option( $key = '', $value = false ) {
+	// First let's grab the current settings
+	$options = get_option( 'edd_settings' );
+
+	// Let's let devs alter that value coming in
+	$value = apply_filters( 'edd_update_option', $value, $key );
+
+	// Next let's try to update the value
+	$options[ $key ] = $value;
```

Comment:
Here's an example of two plugins. One plugin, Plugin 1, is newer and uses the new function. The second plugin, Plugin 2, uses a setting from Plugin 1, and like many extensions it uses the $edd_options global variable without doing an isset check:

https://gist.github.com/chriscct7/b47d1307e503cb464c8d
https://gist.github.com/chriscct7/b544548ef1ab796df770

Testing:
Install and activate plugins 1 and 2
After activating both. Comment out the upper part of plugin 1 (to simulate it updating to using this new function) and uncomment the bottom.